spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/mapping/MongoSimpleTypes.java

```diff
@@ -37,31 +36,38 @@
  *
  * @author Oliver Gierke
  * @author Christoph Strobl
+ * @author Mark Paluch
  */
 public abstract class MongoSimpleTypes {
 
 	public static final Set<Class<?>> AUTOGENERATED_ID_TYPES;
 
 	static {
-		Set<Class<?>> classes = new HashSet<Class<?>>();
+		Set<Class<?>> classes = new HashSet<>();
 		classes.add(ObjectId.class);
 		classes.add(String.class);
 		classes.add(BigInteger.class);
 		AUTOGENERATED_ID_TYPES = Collections.unmodifiableSet(classes);
 
-		Set<Class<?>> simpleTypes = new HashSet<Class<?>>();
+		Set<Class<?>> simpleTypes = new HashSet<>();
 		simpleTypes.add(DBRef.class);
 		simpleTypes.add(ObjectId.class);
 		simpleTypes.add(BsonObjectId.class);
-		simpleTypes.add(CodeWScope.class);
 		simpleTypes.add(org.bson.Document.class);
 		simpleTypes.add(Pattern.class);
 		simpleTypes.add(Binary.class);
 		simpleTypes.add(UUID.class);
 
+		if (ClassUtils.isPresent("org.bson.types.CodeWScope", MongoSimpleTypes.class.getClassLoader())) {
```

Comment:
Looking at the latest driver bits, those changes here do not seem to be necessary any longer. Comparing the `org.bson` packages of `org.mongodb:mongo-java-driver:3.8.0` and `org.mongodb:bson:3.8.0` shows no difference at all.

I think we should wait until #576 is resolved.